fotoapparat/src/main/java/io/fotoapparat/parameter/selector/SelectorFunction.java

```diff
@@ -1,15 +1,14 @@
 package io.fotoapparat.parameter.selector;
 
-import java.util.Collection;
-
 /**
- * Function which selects one element from many.
+ * Function which selects a value based on the input
  */
-public interface SelectorFunction<T> {
+public interface SelectorFunction<Input, Output> {
 
     /**
-     * @return element from given collection.
+     * @return selected value.
      */
-    T select(Collection<T> items);
+    Output select(Input input);
```

Comment:
I tried doing something like this at first, but had some obstacles. `Selectors.firstAvailable` would need to work for both types of `SelectorFunction`, so they would need to be abstract classes extending the same base abstract class. But since `select` is dependent on the "input" type, it's hard to make this work:
```
abstract class SelectorFunction<T> {
  public T select(Object input);
}
abstract class RangeSelectorFunction extends SelectorFunction<T> {
  // does not actually override
  public T select(Range<T> input);
}
abstract class CollectionSelectorFunction extends SelectorFunction<T> {
  // does not actually override
  public T select(Collection<T> input);
}
```
I also tried something like:
```
abstract class SelectorFunction<Input, T> {
  public T select(Input input);
}
abstract class RangeSelectorFunction<T> extends SelectorFunction<Range<T>, T> {
  public T select(Range<T> input);
}
abstract class CollectionSelectorFunction<T> extends SelectorFunction<Collection<T>, T> {
  public T select(Collection<T> input);
}
```
But this also did not work. I think the problem was that Java doesn't consider `RangeSelectorFunction<Integer> to be a subclass of SelectorFunction<Input, Integer>`, but I don't remember the exact error message now.